mistral/mistral.py

```diff
@@ -0,0 +1,287 @@
+# Copyright © 2023 Apple Inc.
+
+import argparse
+from dataclasses import dataclass
+import json
+from pathlib import Path
+from typing import Optional, Tuple, List
+from sentencepiece import SentencePieceProcessor
+
+import mlx.core as mx
+import mlx.nn as nn
+from mlx.utils import tree_map, tree_unflatten
+
+
+@dataclass
+class ModelArgs:
+    dim: int
+    n_layers: int
+    head_dim: int
+    hidden_dim: int
+    n_heads: int
+    n_kv_heads: int
+    norm_eps: float
+    vocab_size: int
+
+
+class RMSNorm(nn.Module):
+    def __init__(self, dims: int, eps: float = 1e-5):
+        super().__init__()
+        self.weight = mx.ones((dims,))
+        self.eps = eps
+
+    def _norm(self, x):
+        return x * mx.rsqrt(x.square().mean(-1, keepdims=True) + self.eps)
+
+    def __call__(self, x):
+        output = self._norm(x.astype(mx.float32)).astype(x.dtype)
```

Comment:
Good question.. I didn't dig into it. Let me see if it passes my tests with using our RMSNorm + float32

Reply:
The other difference is we prescale the input and `sum` rather than `mean` the input.  But I think the `float32` casting is more important as I get more different results when I remove that and do it all in fp16.